restbridge/RestBridge.cpp

```diff
@@ -33,128 +34,126 @@
-
-        server::options options( handler );
-        std::stringstream port;
-        port << port_;
-        // Create http server
-        httpServer_.reset( new server( options.address( hostname_ ).port( port.str( ))));
-
-        RBDEBUG << "HTTP Server runnning on " << hostname_ << ":" << port_ << std::endl;
-        RBDEBUG << "ZeroEq Publisher Schema : " << publisherSchema << std::endl;
-        RBDEBUG << "ZeroEq Subscriber Schema: " << subscriberSchema << std::endl;
-        serverRunning_ = true;
-        // Start processing http requests
-        httpServer_->run();
-    }
+private:
+    std::unique_ptr< Server > _server;
+    std::unique_ptr< RequestHandler > _handler;
+};
+}
 
-    void stop()
-    {
-        if( !thread_ || !serverRunning_ )
-            RBTHROW( std::runtime_error( "HTTP server is not running, cannot stop it" ));
-        // Stop http server hosted by thread_
-        httpServer_->stop();
-
-        // Wait until the http server is stopped
-        std::unique_lock<std::mutex> lock(mutex_);
-        // Stop listening thread
-        thread_->join();
-        serverRunning_ = false;
-        RBDEBUG << "HTTP Server on " << hostname_ << ":" << port_
-                << " is stopped" << std::endl;
-    }
+namespace
+{
+static bool _hasRestParameter( const int argc, char* argv[] )
+{
+    for( int i = 0; i < argc; ++i  )
+        if( std::string( argv[i] ) == "--rest" )
+            return true;
+    return false;
+}
 
-    boost::scoped_ptr< boost::thread > thread_;
-    boost::scoped_ptr< server > httpServer_;
-    std::mutex mutex_;
-    std::atomic<bool> serverRunning_;
-    std::string hostname_;
-    uint16_t port_;
-    std::string schema_;
-};
 }
 
-RestBridge::RestBridge( const std::string& hostname, const uint16_t port )
-    : _impl( new detail::RestBridge( hostname, port ))
+std::unique_ptr< RestBridge > RestBridge::parse(
+    const zeq::Publisher& publisher, int argc, char* argv[] )
 {
+    if( _hasRestParameter( argc, argv ))
+        return std::unique_ptr< RestBridge >( new RestBridge( argc, argv,
+                                                          publisher.getURI( )));
+    return nullptr;
 }
 
-RestBridge::RestBridge( const int argc, const char** argv )
-    : _impl( new detail::RestBridge( argc, argv ))
+std::unique_ptr< RestBridge > RestBridge::parse( const int argc, char* argv[] )
+{
+    if( _hasRestParameter( argc, argv ))
+        return std::unique_ptr< RestBridge >( new RestBridge( argc, argv,
+                                                              zeq::URI( )));
+    return nullptr;
+}
+
+std::unique_ptr< RestBridge > RestBridge::create( const int argc, char* argv[] )
+{
+    return std::unique_ptr< RestBridge >( new RestBridge( argc, argv,
+                                                          zeq::URI( )));
+}
+
+std::string RestBridge::getHelp()
+{
+    static const std::string help(
+        "  --rest [host][:port]: Enable the REST bridge. Optional parameters\n"
```

Comment:
Schema is required to match the Servus URI RFC3986 requirements. "--rest [schema][host][:port]" instead of "--rest [host][:port]"